src/lib/components/BarcodeScanner/BarcodeScanner.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+import { Container, Divider, Segment } from 'semantic-ui-react';
+import Overridable from 'react-overridable';
+import PropTypes from 'prop-types';
+
+class BarcodeScanner extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedDeviceId: null,
+    };
+
+    this.scanner = null;
```

Comment:
We should not have this here, we are adding a dependency to the parent component. 

The `stopScanner()` should receive no params, as it is a callback to the parent cmp.
In the parent cmp, the stopScanner func should:
```
stopScanner:
  this.stopScanner(this.scanner);
```